skia/RNDeviceInfoModule.cpp

```diff
@@ -0,0 +1,370 @@
+/*
+ * Copyright (C) 1994-2023 OpenTV, Inc. and Nagravision S.A.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "ReactSkia/utils/RnsLog.h"
+#include <cxxreact/JsArgumentHelpers.h>
+
+#include "RNDeviceInfoModule.h"
+
+using namespace folly;
+
+namespace facebook {
+namespace xplat {
+
+RNDeviceInfoModule::RNDeviceInfoModule() { }
+
+auto RNDeviceInfoModule::getConstants() -> std::map<std::string, folly::dynamic> {
+  return {
+    {"deviceId", "nagradeviceId"},
+    {"bundleId", "nagrabundleId"},
+    {"systemName", "nagrasystem"},
+    {"systemVersion", "nagraVersion"},
+    {"appVersion","nagrappVersion"},
+    {"buildNumber","nagrabuildNumber"},
+    {"isTablet",false},
+    {"appName","nagraAppname"},
+    {"brand","nagraBrand"},
+    {"model","nagarModel"},
+    {"deviceType","nagraDeviceType"}
+  };
+}
+
+std::string RNDeviceInfoModule::getName() {
+  return "RNDeviceInfo";
+}
+
+auto RNDeviceInfoModule::getMethods() -> std::vector<Method> {
+  return {
+    Method(
+        "isEmulator",
+        [this] (dynamic args, Callback resolveCb , Callback rejectCb) {
+          RNS_LOG_NOT_IMPL;
+          RNS_UNUSED(this);
+          dynamic result = false;
+          resolveCb({result});
+          return;
+        }),
+    Method(
+        "isEmulatorSync",
+        [] () -> bool {
+          RNS_LOG_NOT_IMPL;
+          return false;
+        },SyncTag),
+    Method(
+        "getFontScale",
+        [this] (dynamic args, Callback resolveCb, Callback rejectCb) {
+          RNS_LOG_NOT_IMPL;
+          RNS_UNUSED(this);
+          dynamic result = 1.0;
+          resolveCb({result});
+          return;
+        }),
+    Method(
+        "getFontScaleSync",
+        [] () -> float {
+          RNS_LOG_NOT_IMPL;
+          return 1.0;
+        },SyncTag),
+    Method(
+        "isPinOrFingerprintSet",
+        [this] (dynamic args, Callback resolveCb, Callback rejectCb) {
+          RNS_LOG_NOT_IMPL;
+          RNS_UNUSED(this);
+          dynamic result = false;
+          resolveCb({result});
+          return;
+        }),
+    Method(
+        "isPinOrFingerprintSetSync",
+        [] () -> bool {
+          RNS_LOG_NOT_IMPL;
+          return false;
+        },SyncTag),
+    Method(
+        "getCarrier",
+        [this] (dynamic args, Callback resolveCb, Callback rejectCbr) {
+          RNS_LOG_NOT_IMPL;
+          RNS_UNUSED(this);
+          dynamic result = "Nagra";
+          resolveCb({result});
+          return ;
+        }),  
+    Method(
+        "getCarrierSync",
+        [] () -> std::string {
+          RNS_LOG_NOT_IMPL;
+          return "Nagra";
+        },SyncTag),
+    Method(
+        "getTotalDiskCapacity",
+        [this] (dynamic args, Callback resolveCb, Callback rejectCb) {
+          RNS_LOG_NOT_IMPL;
+          RNS_UNUSED(this);
+          dynamic result = 536870912000;
```

Comment:
hardcoded values can we define them using macro 
to avoid modifications in two diff APIs in future

Reply:
The reason we didn't go with Macro constants because this is  just a dummy implementation. If we add macro for every constants, we will end-up adding macro for many constants which will be removed later ( when we actually implement the methods). Moreover these constants are not re-used.